server/routes/createGoal/createGoalController.ts

```diff
@@ -75,6 +77,8 @@ export default class CreateGoalController {
     const targetDate = parseDate(req, 'targetDate')
     req.session.addStepForm = { ...req.body, targetDate }
     const { addStepForm } = req.session
+    const { addStepForms } = req.session
+    addStepForms.push(req.session.addStepForm)
```

Comment:
I think we only want to push to the array after the form has been validated
As you have it I _think_ you could submit an invalid form and it would get added to the array on L81, before we validate it
The validation fails, we re-render the page, and the user submits again
The new form will be added to the array, so now there are 2 elements in the array, even though its the same StepForm

I think you'd want to move L81 to L88

What do you think?